### DISCUSSION TITLE
How to fix subtitles staying on screen until new ones show up?

### DISCUSSION THREAD

**Discussion by dillfrescott** [openai/whisper#2210] [Q&A] [answered]
This is happening to me and I dont know how to solve it other than having to manually change the timestamps for every utterance.

**Comment by glangford** [comment]
whisper doesn't do this (there is no internal option to have subtitles linger). You could write code to post-process the output to look at pairs of subtitles, and extend the end time of the first subtitle to stay on screen until the start of the second.

**Reply by dillfrescott** [reply]
No, its lingering them and I want it to **not** do that.

**Reply by dillfrescott** [reply]
It happens every time

**Reply by glangford** [reply, marked as answer]
Ah....ok sorry, I misunderstood. Do you use
`--word_timestamps True`

**Reply by dillfrescott** [reply]
Oh, no I didnt even know that was an option. Ill try it soon!

**Reply by glangford** [reply]
Words are individually timestamped with this option, and then subtitles are generated using the timing of the first and last words in the subtitle block. Generally works very well.

**Reply by dillfrescott** [reply]
Yup that seemed to have solved it! Thank you so much!!!

**Comment by creativeidiot123** [comment]
> Ah....ok sorry, I misunderstood. Do you use
> `--word_timestamps True`

Unrelated but i need your opinion on this please

https://github.com/openai/whisper/discussions/2212

I use your spacy script to deal with this. But it only works with word timestamps enabled. And enabling that gives me cuda triton error as I'm using windows. 

@glangford

**Reply by glangford** [reply]
> But it only works with word timestamps enabled. And enabling that gives me cuda triton error as I'm using windows.

@ankitgurua The spacy code requires word timestamps, there is no way around that. This is because when the subtitles are reordered, word timestamps are used to set the start and stop times of the subtitle block.

I suggest you create a new discussion thread with the specifics of your triton error (including your version of windows, what you have tried already, etc.)

**Reply by creativeidiot123** [reply]
@glangford I was thinking of trying to get spacy code to work with WhisperX. Considering it also got word timestamp enabled but it doesn't give that triton error for windows users. WhisperX is also faster and have higher accuracy then Whisper itself. 

But Spacy file currently don't work with the JSON from WhisperX.

**Reply by glangford** [reply]
@ankitgurua I don't use whisperX. You would have to look at the json format whisperX uses and modify the code accordingly.
